openpype/hosts/unreal/plugins/publish/collect_remove_marked.py

```diff
@@ -0,0 +1,24 @@
+import pyblish.api
+
+
+class CollectRemoveMarked(pyblish.api.ContextPlugin):
+    """Remove marked data
+
+    Remove instances that have 'remove' in their instance.data
+
+    """
+
+    order = pyblish.api.CollectorOrder + 0.499
+    label = 'Remove Marked Instances'
+
+    def process(self, context):
+
+        self.log.debug(context)
+        # make ftrack publishable
+        instances_to_remove = []
+        for instance in context:
+            if instance.data.get('remove'):
+                instances_to_remove.append(instance)
+
+        for instance in instances_to_remove:
+            context.remove(instance)
```

Comment:
This logic is a duplicate of other hosts.

In [StandalonePublisher](https://github.com/pypeclub/OpenPype/blob/7017410173647da9a122e0659fda2c7af5593c80/openpype/hosts/standalonepublisher/plugins/publish/collect_remove_marked.py) and [Maya](https://github.com/pypeclub/OpenPype/blob/97cd765363961d8a2e756c17cf0a91aba95f3c95/openpype/hosts/maya/plugins/publish/collect_remove_marked.py). Would there be a way to unify this behavior?

E.g. make a single plug-in in global and add `hosts = ["maya", "standalonepublisher", "unreal"]` filter or alike?

Reply:
For what it's worth. I think the logic in [StandalonePublished CollectRemoveMarked](https://github.com/pypeclub/OpenPype/blob/7017410173647da9a122e0659fda2c7af5593c80/openpype/hosts/standalonepublisher/plugins/publish/collect_remove_marked.py#L19-L21) isn't even safe because it's removing instances whilst iterating the same list?

More info as to why [that is not safe](https://stackoverflow.com/a/6260097/1838864).